Fixes #33319 - don't run fix-stories on every postinstall

evgeni opened the pull request:
<!---

Thank you for contributing to The Foreman, please read the
[following guide](https://www.theforeman.org/contribute.html), in short:

* [Create an issue](https://projects.theforeman.org/projects/foreman/issues)
* Reference the issue via `Fixes #1234` in the commit message
* Prefer present-tense, imperative-style commit messages
* Mark all strings for translation, see [1]
* Suggest prerequisites for testing and testing scenarios following example above.
* Prepend `[WIP]` for work in progress to prevent bots from triggering actions
* Be patient, we will do our best to take a look as soon as we can
* Explain the purpose of the PR, attach screenshots if applicable
* List all prerequisites for testing (e.g. VMware cluster, two smart proxies...)
* Reviewers often use extensive list of items to check, have a look prior submitting [2]
* Be nice and respectful

1: https://projects.theforeman.org/projects/foreman/wiki/Translating#Translating-for-developers
2: https://github.com/theforeman/foreman/blob/develop/developer_docs/pr_review.asciidoc
-->

evgeni committed:
Fixes #33319 - don't run fix-stories on every postinstall

ekohl commented:
In e7c0ad82a2f327e105f3ceae62b2b6721783ae5a this was added. It'd be helpful to have a quick explanation why it's no longer needed and if any developer documentation about using stories should be updated.

evgeni commented:
before:

```
added 1617 packages from 759 contributors and audited 1641 packages in 367.97s
added 4251 packages from 2660 contributors and audited 4259 packages in 1012.048s
```

after:

```
added 4251 packages from 2660 contributors and audited 4259 packages in 419.668s
```

note: the timing *might* be off as different jenkins nodes are differently fast

but the interesting thing is: it still installs *all* >4000 packages, so it might not solve all our problems, but still the cleaner approach

evgeni commented:
> In [e7c0ad8](https://github.com/theforeman/foreman/commit/e7c0ad82a2f327e105f3ceae62b2b6721783ae5a) this was added. It'd be helpful to have a quick explanation why it's no longer needed and if any developer documentation about using stories should be updated.

I agree, but I'd need someone like @laviro to explain to me what I am doing here (and why it was needed before) :D

tbrisker commented:
> audited 4259 packages

not directly related, but we can run with `--no-audit` on CI. Might save some time as well.

Ron-Lavi commented:
> I agree, but I'd need someone like @laviro to explain to me what I am doing here (and why it was needed before) :D

looking at it more closely, it seems we need it for using stories [until the migration to webpack-4 will be done.](https://github.com/theforeman/foreman/commit/e7c0ad82a2f327e105f3ceae62b2b6721783ae5a#diff-ab3eaebeffa5420102a2e4d5ce23d143e44277d6bfc632c0085ffcdd7e5d84ceR6)
do stories still work without it now?

I like what you did in this PR with moving the script to another place,
IMHO we don't want it in each post-install as not everyone using it and it makes everything longer,
and if someone wants to use storybook documentation, just make sure everything is installed

cc @amirfefer @MariaAga, any other thoughts?

Ron-Lavi commented on package.json:
```diff
@@ -14,9 +14,10 @@
     "test:current": "tfm-test --watch",
     "publish-coverage": "tfm-publish-coverage",
     "stories": "tfm-stories",
+    "fix-stories": "./script/npm-fix-foreman-stories.sh",
```
how about `install:stories` ?
```suggestion
    "install:stories": "./script/npm-fix-foreman-stories.sh",
```

evgeni replied:
does that have some special meaning? or just convention?

Ron-Lavi replied:
Just a convention

evgeni replied:
We could also move that to the actual `stories` entry, and check for the existence of `node_modules/@theforeman/stories/node_modules` in `npm-fix-foreman-stories.sh`, so that it only gets called once on the first `stories` invocation?

Ron-Lavi replied:
+1 for that

evgeni replied:
Updated!

evgeni commented:
> > audited 4259 packages
> 
> not directly related, but we can run with `--no-audit` on CI. Might save some time as well.

good point, https://github.com/theforeman/jenkins-jobs/pull/87

evgeni commented:
And @laviro, you're right `npm run stories` doesn't work on a fresh clone without running `fix-stories`.

Ron-Lavi commented on script/npm-fix-foreman-stories.sh:
```diff
@@ -9,5 +9,9 @@ if [ "$NODE_ENV" = "production" ]; then
   exit 0
 fi
 
+if [ -d node_modules/@theforeman/stories/node_modules ]; then
```
Nice!

amirfefer reviewed (approved):
Thanks @evgeni 👍
